lib/Network/Linklater/Types.hs

```diff
@@ -122,6 +139,22 @@ commandOfParams params = do
       Just value -> Right value
       Nothing -> Left ("paramOf: no key: " <> show key)
 
+eventOfBody :: Data.Aeson.Object -> Either String Event
+eventOfBody obj = do
+  user <- User <$> valueOf "user"
+  channel <- valueOf "channel"
+  eventType <- valueOf "type"
+  case eventType of
+    "link_shared" -> do
+      let links = LinkShared $ obj ^.. ix "links" . values . _Object . ix "url" . _String 
+      Event <$> pure links <*> pure user <*> pure channel
```

Comment:
i think `pure (Event links user channel)` might be easier to read here (same note for a couple lines down)